### ISSUE TITLE
"Are You Sure?" popup before adding a Note for the very first time

### ISSUE THREAD

**Issue by Tex2002ans** [streetcomplete/StreetComplete#5679]
### Use case

When a user:

- Is completely new
   - 0 quests completed, 0 Notes in StreetComplete, etc.
- Leaves a Note in StreetComplete for the very first time

it would be nice to have a window that informs + asks you "Are you sure? This will be shared with the public!" beforehand.

### Proposed Solution

After a user follows these steps:

1. Press+Hold on map
2. Choose "Create new note"
3. Type something in, then press the OK checkmark in lower-right corner.

For the very first time, pop up a warning like:

> "This Note will be shared to the public. Are you sure?"
>
> YES / NO
> [ ] Do not show this again.

If they:

- Choose "Yes".
  - Continue just like it does now.
- Choose "No".
  - Bring you back to the Note box, so you can decide what to do with it.
  - (Like copy/paste text elsewhere or edit it if needed.)
- If "Do not show this again" is...
   - CHECKED, then this popup will never appear again.
   - UNCHECKED, the popup can appear next time too.
      - Similar to the current popup for "StreetComplete is meant for surveying. Are you SURE you verified this feature in-person?"

- - -

Intermediate/Advanced users will just:

- CHECK the box once.
- Never see popup again.

Beginner users will:

- Get stopped/informed.
- Check/Uncheck the box as needed.
   - CHECK = Never be bothered by popup again.
   - UNCHECK = Can leave themselves a future "warning" just in case.

- - -

#### Use-case 1

Last month, **I was completely new to OSM+StreetComplete**.

I was surveying addresses/streets—I made a few mistakes + had no idea how to Undo yet.

**I thought Notes were local-on-device** only, so I left 2 or 3 "notes to myself", thinking I would get back to them when I got home.

About a week later, I got an email because an OSM user commented on my Notes. The Notes were:

- In a bad/imprecise location, so they weren't sure which houses it applied to.
- Low-quality + not descriptive.
   - My Note just said something cryptic like "1234" where I accidentally typed "4234" in the address quest.

Luckily, I already fixed it in OSM a few days after I "left the note to myself"... but it could have been embarrassing if there was "personal info/photos" in there.

(Honestly, I was thinking it was like Organic Maps's bookmarks, which are local-only.)

#### Alternate Solution 1: Bookmarks

Introduce a local "bookmarks" feature, similar to Organic Maps. This would stay on-device only.

(But this would probably clutter the UI/menus, or still be confusing for new users. "What is the difference between Notes/Bookmarks + 'Uhhh...'?" And would just shift the problem elsewhere.)

**Comment by mnalis** [comment]
Thanks for the feedback! It is very valuable, as each one of us can a newbie only once :smiley_cat:  
I can see how newcomer might be taken by surprise there by notes being public.

To be fair, every Note dialog does say _**"You can leave a note to let other mappers know about an issue here..."**_ which should make it clear that it is going to be public. However, I also get that many users often don't read _"small print"_...

However, the proposed solution seems somewhat complex to me. I think just a confirmation popup _**"Notes are always shared to the public! Do you want to proceed?"**_ the **first** time when the user uploads their first Note might be helpful.

Alternatively (and more simple), perhaps just a menu entry could be renamed from `Create new note` to `Create new public note`.

**Comment by Tex2002ans** [comment]
Yeah, all those sound like reasonable solutions too. :)

Just wanted to put this suggestion out there, then others can decide what the best course of action might be.

> To be fair, every Note dialog does say "You can leave a note to let other mappers know about an issue here..." which should make it clear that it is going to be public. However, I also get that many users often don't read "small print"...

Yeah, I see that... NOW! But not within the first few days of StreetComplete! :P

I think at the time I was coming as a light user of Organic Maps, so just thought: "offline bookmarks" under a different name!

- - -

**Side Note:** And, I think another situation that may have happened in the very beginning:

- By default, StreetComplete is set to "Upload answers automatically" = ON.

I was leaving these Notes for the very first time... so I thought since my StreetComplete was "not connected to wifi"/"offline", no way these "personal notes" can make it into the public map "without my explicit approval" / "pushing the upload button" type of thing.

So when StreetComplete began auto-uploading everything, I also panicked.

(I've now since set "Upload answers" -> "Wifi Only".)

Anyway, within a week, I figured it all out... But those initial few days bit me. :)

**Comment by rhhsm** [comment]
> 
> Intermediate/Advanced users will just:
> 
>     * CHECK the box once.
> 
>     * Never see popup again.

There are many popups that I still get to see after 1000+ activity days that I'd love to get rid of, so a checkbox that will make them go away for ever would be very welcome... It could be an award after making x contributions too (to express that "now you're not a newbie any more and we trust that you read these popups").

> Introduce a local "bookmarks" feature, similar to Organic Maps. This would stay on-device only.
> 
See #1992

**Comment by matkoniecz** [comment]
> There are many popups that I still get to see after 1000+ activity days that I'd love to get rid of

Are there issues proposing doing this? If no, then feel free to open such!

> It could be an award after making x contributions too (to express that "now you're not a newbie any more and we trust that you read these popups").

This, and also if someone got into habit of dismissing them and ignoring them... In such case showing them will not help anyway, so there is no real loss.

**Comment by Tex2002ans** [comment]
> There are many popups that I still get to see after 1000+ activity days that I'd love to get rid of, so a checkbox that will make them go away for ever would be very welcome... [...]

Create the issue and I'll take a look.

I can't really think of any popups that have been "bothering" me. The only one I can recall happening often is the "Are you sure?" because you are too far away, but it's easily dismissable per session.

(And it's something I can see in the distance, like CLEARLY that's an asphalt parking lot. No need for me to go walking all the way over there! :P)

>> Introduce a local "bookmarks" feature, similar to Organic Maps. This would stay on-device only.
> 
> See #1992

Thanks. I'll check it out.

Right now, if I need to leave "private notes", I just leave myself "bookmarks" on Organic Maps.

For example, as I was out mapping... I:

- Noticed a few weirdly "duplicate" street names printed on small, residential streets.
    - (Most likely they were an accidentally split road.)
    - So I mark them in Organic Maps so I can investigate on OSM.org later and take a much closer look at the ways/tags. Then I merge them into a single way.
- Noticed an odd street name.
   - Like one was called "16".
   - No street signs or anything, but it was a little backroad to connect garages.
- Get asked "asphalt" quest on someone's driveway.
   - Whoops, it was accidentally tagged as a "service road".
   - When I get back to computer, I change it from "service"->"driveway".

Not anything big that I would want to bother another mapper about, but just something to do very small cleanup in. :)

**Comment by westnordost** [comment]
Hm, I have just read this issue. Some of it was for what is now #5683. Regarding the original request, I would say, this is a will-not-fix.
As mnalis pointed out, the message shown clearly states that this is a note for other mappers. If a user does not read, it usually does not help to pad the app with even more text and confirmation dialogs.
A solution could be to change the text of the button from "`Leave note`" to "`Leave public note`", but button texts should be short. You never know in which abstruse language this will end up incredibly long - German for example: "`Öffentlichen Hinweis hinterlassen`".
